packages/bolt-integration/src/bolt-button-strategy.ts

```diff
@@ -100,20 +116,67 @@ export default class BoltButtonStrategy implements CheckoutButtonStrategy {
         boltButtonContainer.setAttribute('style', 'display:none');
         boltButtonContainer.setAttribute('data-tid', 'product-page-checkout-wrapper');
 
-        boltButtonObject.setAttribute('data', this.getBoltObjectData(paymentMethod));
+        boltButtonObject.setAttribute('data', this.getBoltObjectData(paymentMethod, style));
         boltButtonObject.setAttribute('class', 'bolt-product-checkout-button');
 
         boltButtonContainer.append(boltButtonObject);
         container.innerHTML = '';
         container.append(boltButtonContainer);
     }
 
-    private getBoltObjectData(paymentMethod: BoltPaymentMethod): string {
+    private getBoltObjectData(
+        paymentMethod: BoltPaymentMethod,
+        style?: BoltButtonStyleOptions,
+    ): string {
         const { initializationData, config } = paymentMethod;
         const { publishableKey, developerConfig } = initializationData || {};
 
         const domainUrl = this.boltScriptLoader.getDomainURL(!!config.testMode, developerConfig);
+        const buttonHeight = this.getButtonHeight(style?.size);
+        const buttonBorderRadius = this.getButtonBorderRadius(style?.shape, buttonHeight);
+        const urlParameters = stringify({
+            publishable_key: publishableKey,
+            variant: 'ppc',
+            ...(buttonHeight && { height: buttonHeight }),
+            ...(buttonBorderRadius && { border_radius: buttonBorderRadius }),
+        });
+
+        return `https://${domainUrl}/v1/checkout_button?${urlParameters}`;
```

Comment:
[Optional to change]

Here is another way how this piece can be done:
```
return stringifyUrl({
    currentUrl:  `https://${domainUrl}/v1/checkout_button`,
    query: {
        publishable_key: publishableKey,
        variant: 'ppc',
        height: buttonHeight,
        border_radius: buttonBorderRadius,
    },
});
```

Reply:
done